##### Exam 1 Answers by Javier López Rodríguez  (javier.lopez.rodriguez@alumnos.upm.es)

### Problem 2

ls shows a list of the directory contents.

Option -l shows the information in a long listing format, in which the ownership appears in the third (user) and fourth (group) fields of each row.

Option -h shows file sizes in a human-readable format.

In [1]:
ls -lh

total 3.6M
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 12:01  conv.txt
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 17:45  Data_Only.csv
-rw-rw-r-- 1 osboxes osboxes  147 Sep  3 12:01  DISCLAIMER
-rw-rw-r-- 1 osboxes osboxes 4.5K Sep  3 17:45  Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes 5.5K Sep  3 12:01 'Exam Week 1.ipynb'
-rw-rw-r-- 1 osboxes osboxes 3.9K Sep  3 12:01 'Exam Week 2.ipynb'
-rw-rw-r-- 1 osboxes osboxes 4.2K Sep  3 12:01  Germplasm.tsv
-rw-rw-r-- 1 osboxes osboxes  141 Sep  3 12:01  how_to_convert.txt
-rw-rw-r-- 1 osboxes osboxes 1.1K Sep  3 12:01  LICENSE
-rw-rw-r-- 1 osboxes osboxes  637 Sep  3 12:01  LocusGene.tsv
-rw-rw-r-- 1 osboxes osboxes 1.2M Sep  3 12:01  Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes  743 Sep  3 12:01  README.md


The permissions of this file are shown in the first field, "-rw-rw-r--".

Characters 2-4 refer to the permissions of the user. A letter (r, w and x) shows if that user is permitted to read, write and execute the file, respectively. A dash (-) indicates a lack of permission. Therefore, "rw-" indicates that the user can read and write, but not execute.

Characters 5-7 show the permissions of the group, the same way as explained above. "rw-" indicates that the group can read and write, but not execute.

Characters 8-10 show the permissions of all other users. "r--" indicates that anyone can read the file, but not write or execute.

### Problem 3

head -n outputs the first n lines of a file. In this case, head -1 shows the first line (the header).

In [2]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


### Problem 4

wc -l counts the number of lines of a file.

In [3]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


### Problem 5

tail -n +NUM outputs every line starting with the NUM-th line. In this case, tail -n +2 starts with the 2nd line, therefore skipping the header.

The output of tail gets redirected (>) to the new file.

For the verification, wc -l outputs the number of lines of each file, as expected (7216 in the original, 7215 in Data_Only.csv)

In [4]:
# Writing only the data lines into the new file.
tail -n +2 Locus_Germplasm_Phenotype_20130122.txt > Data_Only.csv

# Checking that the new file has one less line than the original file.
echo "Number of lines of the original file:"
wc -l Locus_Germplasm_Phenotype_20130122.txt
echo "Number of lines of the new file:"
wc -l Data_Only.csv

Number of lines of the original file:
7216 Locus_Germplasm_Phenotype_20130122.txt
Number of lines of the new file:
7215 Data_Only.csv


### Problem 6

### Problem 7

### Problem 8

### Problem 9

### Problem 10